common/v2/features/SwapAssets/stateFactory.tsx

```diff
@@ -204,7 +215,10 @@ const SwapFlowFactory: TUseStateReducerFactory<SwapState> = ({ state, setState }
         }).toString(),
         fromAmountError: '',
         toAmountError: '',
-        swapPrice: price
+        swapPrice: price,
+        initialValue: Number(value) * price,
```

Comment:
Quote from @vilsbole :
> Danger! Since we're dealing with 18 decimal values, and price is determined by an external provider this operation is unsafe and lead to an (over|under)flow. All multiplications and divisions MUST (RFC2119 be done using safe operations. In this project we use ethers.js [modified](https://ethereum.stackexchange.com/a/77898/24916) BN.js

Please use `convert` function for those operations